compositor_render/src/transformations/shader/validation.rs

```diff
@@ -254,100 +318,382 @@ fn validate_type_equivalent(
     Ok(())
 }
 
+#[derive(Debug, thiserror::Error)]
+pub enum ConstArraySizeEvalError {
```

Comment:
I thought about it, I don't know how to do it. Unlike structs, array types don't have names. I figured the best we can do is to give information about which field in a struct this problem originated in (in case this array is a part of a struct)